app/src/main/java/com/example/pygmyhippo/user/EventFragment.java

```diff
@@ -0,0 +1,133 @@
+                "hippoparty.png",
+                Event.EventStatus.ongoing
+        );
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+    }
+
+    // populates the view with information
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        View eventBackButtonView = view.findViewById(R.id.u_backButtonToQRView);
+        TextView eventNameView = view.findViewById(R.id.u_eventNameView);
+        TextView eventDateView = view.findViewById(R.id.u_eventDateView);
+        TextView eventTimeView = view.findViewById(R.id.u_eventTimeView);
+        TextView eventOrganizerView = view.findViewById(R.id.u_organizerNameView);
+        TextView eventLocationView = view.findViewById(R.id.u_eventLocationView);
+        TextView eventCostView = view.findViewById(R.id.u_eventCostView);
+        TextView eventAboutTitleView = view.findViewById(R.id.u_aboutEventTitleView);
+        TextView eventAboutDescriptionView = view.findViewById(R.id.u_aboutEventDescriptionView);
+        View eventRegisterButtonView = view.findViewById(R.id.u_registerButton);
+
+        // set with hardcoded values, set the view in the same wy form the event recieved from the constructor
+        event = hardcodeEvent();
+
+        eventNameView.setText(event.getEventTitle());
+        eventDateView.setText(event.getDate());
+        eventTimeView.setText(event.getTime());
+        eventOrganizerView.setText(event.getOrganiserID());
+        eventLocationView.setText(event.getLocation());
+        eventCostView.setText(event.getCost());
+        eventAboutDescriptionView.setText(event.getDescription());
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.user_fragment_event, container, false);
+
+        Button registerButton = view.findViewById(R.id.u_registerButton);
+
+        // with a check, check to see if event has been registered for, if so, register (set text
+        // to checkmark, add to lists and stuff, if event has not been register for, set text back
+        // to register and remove from lists and stuff
+        registerButton.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view) {
+                // TODO: add actual database stuff here, where user is added into the events list
+                // for now, this is hardcoding to figure out structure of entrant
+                // this part, the entrant information would be passed to the event but hardcode
+                entrant = new Entrant(
+                        "123",
+                        Entrant.EntrantStatus.invited
+                );
+                event.addEntrant(entrant);
+
```

Comment:
In the future if my PR ever goes through, we might have to add code here to add entrants to the database with an event ID. Or I'm thinking I may have to change my implementation if storing events in Firebase keeps the list data intact.

Reply:
definietly, this is just the hardcoded example code right now, will add a comment clarifying though!